src/hypertable_cache.c

```diff
@@ -129,24 +130,54 @@ ts_hypertable_cache_invalidate_callback(void)
 
 /* Get hypertable cache entry. If the entry is not in the cache, add it. */
 TSDLLEXPORT Hypertable *
-ts_hypertable_cache_get_entry(Cache *cache, Oid relid)
+ts_hypertable_cache_get_entry(Cache *const cache, const Oid relid, const bool missing_ok)
 {
+	Hypertable *ht;
+
 	if (!OidIsValid(relid))
-		return NULL;
+		if (missing_ok)
+			return NULL;
+		else
+			ereport(ERROR, (errcode(ERRCODE_UNDEFINED_OBJECT), errmsg("invalid Oid")));
```

Comment:
Nit, just realized there's an `ERRCODE_UNDEFINED_TABLE`, which is probably more suitable here. A common PG error message in these cases is also `relation with OID %u does not exist`

Reply:
I believe this comment is suitable for https://github.com/timescale/timescaledb/pull/1658/files#diff-3086b70871dfb929dfe95db4f60ef7b5R150 and I updated it with:
```
(errcode(ERRCODE_UNDEFINED_TABLE), errmsg("OID %u does not refer to a table", relid))
```

The line, which you target, is about OID value, which is invalid, i.e., `0`. The error code is correct for this case. I haven't found a common error message in PG.